pkg/api/testing/conversion.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testing
+
+import (
+	"testing"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/labels"
+)
+
+// TestSelectableFieldLabelConversions verifies that given resource have field
+// label conversion defined for each its selectable field.
+// fields contains selectable fields of the resource.
+// labelMap maps deprecated labels to their canonical names.
+func TestSelectableFieldLabelConversionsOfKind(t *testing.T, apiVersion string, kind string, fields labels.Set, labelMap map[string]string) {
+	badFieldLabels := []string{
+		"name",
+		".name",
+		"bad",
+		"metadata",
+		"foo.bar",
+	}
+
+	value := "value"
+
+	if len(fields) == 0 {
+		t.Logf("no selectable fields for kind %q, skipping", kind)
+	}
+	for label := range fields {
+		if label == "name" {
```

Comment:
This doesn't affect a set of supported fields of conversion functions. `"name"` should still be supported if it were used in the past. This just warns about the `"name"` being returned in SelectableFields function for given type. It should return a subset of what configuration fuctions support in order to prevent deprecated fields being used in the future.